src/java.desktop/share/classes/javax/swing/plaf/synth/SynthRootPaneUI.java

```diff
@@ -42,6 +42,7 @@
 public class SynthRootPaneUI extends BasicRootPaneUI implements SynthUI {
     private SynthStyle style;
     static final AltProcessor altProcessor = new AltProcessor();
+    static boolean altProcessorInstalledFlag;
```

Comment:
Wouldn't it be easier to install `altProcessor` always in `SynthLookAndFeel.initialize` and to uninstall it in `SynthLookAndFeel.uninitialize` like it's done in `WindowsLookAndFeel`:

https://github.com/openjdk/jdk/blob/master/src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsLookAndFeel.java#L197-L198

Then `altProcessor` would do nothing if the `RootPane.altPress` property isn't set or is `false`.

Requesting the value of `RootPane.altPress` from `UIManager` each time `postProcessKeyEvent` is called is inefficient, so you can store the value in `altProcessor` when look and feel is installed.

If `RootPane.altPress` can be changed dynamically, you can install a `PropertyChangeListener` to `UIManager`.

Reply:
> Wouldn't it be easier to install altProcessor always in SynthLookAndFeel.initialize and to uninstall it in SynthLookAndFeel.uninitialize like it's done in WindowsLookAndFeel:
https://github.com/openjdk/jdk/blob/master/src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsLookAndFeel.java#L197-L198
Then altProcessor would do nothing if the RootPane.altPress property isn't set or is false.

In my initial fix, I added the `altProcessor` handler in `SynthLookAndFeel.initialize` with condition check for GTK L&F. Phil has suggested not to check for GTK L&F instead look for some alternate way like mentioned [here](https://github.com/openjdk/jdk/pull/18992#discussion_r1595782003).

Now I left with few options:
1. Install `altProcessor` handler similar to `WindowsLookAndFeel` implementation but that results in unnecessary installation of `alt handler for Nimbus L&F (derived from Synth L&F)` as well.
2. Add the `RootPane.altPress` condition check before installing the `altProcessor` handler but the value returned for `RootPane.altPress` is always **false** and that left with `altProcessor` handler uninstalled.

So, the handler implementation is moved to SynthRootPaneUI.

Reply:
>Requesting the value of RootPane.altPress from UIManager each time postProcessKeyEvent is called is inefficient, so you can store the value in altProcessor when look and feel is installed.

I guess you are pointing out the code in SynthGraphicsUtils.paintText method where RootPane.altPress value is retrieved from UIManager each time. Storing the value in it's constructor doesn't reflect the correct value and is always `false`.

Reply:
> If RootPane.altPress can be changed dynamically, you can install a PropertyChangeListener to UIManager.

I think it can be changed but is it really required to handle it ?
I mean why does a user change it dynamically ?

Reply:
> In my initial fix, I added the `altProcessor` handler in `SynthLookAndFeel.initialize` with condition check for GTK L&F. Phil has suggested not to check for GTK L&F instead look for some alternate way like mentioned [here](https://github.com/openjdk/jdk/pull/18992#discussion_r1595782003).
> 
> So, the handler implementation is moved to `SynthRootPaneUI`.

Thus, out of Synth-derived Look and Feels, only GTK needs the feature. Can't the listener be installed in `GTKLookAndFeel.initialize`?

At the same time, if you install the listener in `SynthLookAndFeel`, other L&F based on Synth could also use this feature, thus your way is more flexible.

However, `SynthRootPaneUI` is still part of the base class… I still don't understand how it's different from what I'm proposing.

I'd like to avoid keeping a flag whether the listener is installed or not… But there could be no other way since not all Synth-based L&Fs enable hiding mnemonics.

Reply:
> > Requesting the value of RootPane.altPress from UIManager each time postProcessKeyEvent is called is inefficient, so you can store the value in altProcessor when look and feel is installed.
> 
> I guess you are pointing out the code in SynthGraphicsUtils.paintText method where RootPane.altPress value is retrieved from UIManager each time. Storing the value in it's constructor doesn't reflect the correct value and is always `false`.

No, I didn't refer to this particular case. I referred to my suggestion where you always add `altProcessor` as the keyboard listener and keep track of the value of the `RootPane.altPress` property.

Anyway this approach seems to be rejected.

Reply:
> > If RootPane.altPress can be changed dynamically, you can install a PropertyChangeListener to UIManager.
> 
> I think it can be changed but is it really required to handle it ? I mean why does a user change it dynamically ?

Not impossible.

On Windows, there used to be an option to hide or display mnemonics and focus indicators, and this option could be changed while the app is running.

I think there's no option in GNOME, so we shouldn't go out of the way to support dynamic change of the `RootPane.altPress` property.